File: Chapter06/Java/CommonControls/app/src/main/java/com/androidbook/commoncontrols/RadioGroupActivity.java
package com.androidbook.commoncontrols;

import android.os.Bundle;
import android.util.Log;
import android.widget.RadioGroup;

import androidx.activity.ComponentActivity;

public class RadioGroupActivity extends ComponentActivity {
    protected static final String TAG = "RadioGroupActivity";

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.radiogroup);

        RadioGroup radGrp = findViewById(R.id.radGrp);

        int checkedRadioButtonID = radGrp.getCheckedRadioButtonId();

        radGrp.setOnCheckedChangeListener((arg0, id) -> {
            switch (id) {
                case -1:
                    Log.v(TAG, "Choices cleared!");
                    break;
                case R.id.chRBtn:
                    Log.v(TAG, "Chose Chicken");
                    break;
                case R.id.fishRBtn:
                    Log.v(TAG, "Chose Fish");
                    break;
                case R.id.stkRBtn:
                    Log.v(TAG, "Chose Steak");
                    break;
                default:
                    Log.v(TAG, "Huh?");
                    break;
            }
        });
    }
}

File: Chapter06/Kotlin/CommonControls/app/src/main/java/com/androidbook/commoncontrols/RadioGroupActivity.kt
package com.androidbook.commoncontrols

import android.os.Bundle
import android.util.Log
import android.widget.RadioGroup
import androidx.appcompat.app.AppCompatActivity

class RadioGroupActivity: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.radiogroup)

        val radGrp = findViewById<RadioGroup>(R.id.radGrp)

        val checkedRadioButtonID = radGrp.checkedRadioButtonId

        radGrp.setOnCheckedChangeListener {_, id ->
            when (id) {
                -1 -> Log.v(TAG, "Choices cleared!")
                R.id.chRBtn -> Log.v(TAG, "Chose Chicken")
                R.id.fishRBtn -> Log.v(TAG, "Chose Fish")
                R.id.stkRBtn -> Log.v(TAG, "Chose Steak")
                else -> Log.v(TAG, "Huh?")
            }
        }
    }

    companion object {
        const val TAG = "RadioGroupActivity"
    }
}